lib/Parser/List/Vector.pm

```diff
@@ -22,6 +22,9 @@ sub _check {
 			$type = (($type =~ m/^[aeiou]/i) ? "an " : "a ") . $type;
 			$self->{equation}->Error([ "Coordinates of Vectors must be Numbers, not %s", $type ]);
 		}
+		if ($self->context->flag("requireConstantEntries") && !($x->{isConstant})) {
+			$self->{equation}->Error("Coordinates of Vectors must be constant");
+		}
```

Comment:
You don't need to check each coordinate separately, as the vector's `isConstant` flag will already be set by the time the `_check` subroutine is called, so I think you can just do

``` perl
	$self->{equation}->Error("Coordinates of Vectors must be constant");
	  if $self->context->flag("requireConstantEntries") && !($self->{isConstant};
```

after the `foreach` loop.